File: src/main/java/com/dingdo/common/exception/CheckExceptionHandler.java
package com.dingdo.common.exception;

import com.dingdo.model.msgFromCQ.ReplyMsg;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class CheckExceptionHandler {

    @ExceptionHandler(CheckException.class)
    public ReplyMsg handleBusinessException(CheckException e) {
        ReplyMsg replyMsg = new ReplyMsg();
        replyMsg.setReply(e.getmessage());
        return replyMsg;
    }
}

File: src/main/java/com/dingdo/common/exception/CheckExceptionHandler.scala
package com.dingdo.common.exception

import org.springframework.web.bind.annotation.{ExceptionHandler, RestControllerAdvice}

/**
 * @date 2020/12/2 10:50
 * @author slamacraft 
 * @since JDK 1.8
 * @version 1.0
 */
@RestControllerAdvice
class CheckExceptionHandler {

  @ExceptionHandler(Array(classOf[CheckException]))
  def handleBusinessException(e: CheckException): String = e.getMessage

}
